toolchain/sem_ir/file.cpp

```diff
@@ -44,10 +44,11 @@ auto TypeInfo::Print(llvm::raw_ostream& out) const -> void {
 File::File(SharedValueStores& value_stores)
     : value_stores_(&value_stores),
       filename_("<builtins>"),
-      // Builtins are always the first IR, even when self-referential.
-      cross_reference_irs_({this}),
       type_blocks_(allocator_),
       inst_blocks_(allocator_) {
+  CARBON_CHECK(cross_reference_irs_.Add(this) == CrossReferenceIRId::Builtins)
+      << "Builtins must be the first IR, even if self-referential";
```

Comment:
I would prefer that we don't put code with side-effects in a `CARBON_CHECK`: as a reader I expect that the `CHECK` expression is only computing the value that we're validating and will skip over it when looking for the behavior of the function.

Reply:
Done